### DISCUSSION TITLE
A way to squash history?

### DISCUSSION THREAD

**Discussion by p-hebert** [theory/sqitch#620] [Ideas]
Hi there,

I was wondering if there was a way / a plan to introduce a way to squash previous migrations history. We currently have over 200 migrations and counting. Some of them date from two years ago and do not serve much use for us to keep around any longer.

Is there an easy way to rewrite (squash) a subset of earlier commits in the sqitch plan and deploy the rewritten history to a database that has already the history of the previous changes?

Have a great day!
Cheers,
Philippe

**Comment by theory** [comment]
Hi @philippefutureboy. There is not a feature for this built-in to Sqitch. The pattern for this sort of thing is generally to [dump the database and create a new plan](https://github.com/sqitchers/sqitch/wiki/How-do-I-get-started-with-an-existing-database%3F). That FAQ item assumes you're starting from an unmanaged database, but in your case, if you're using the same plan name and URL, you'd want to take the additional step of removing all the changes from the registry before deploying the new single change. If it's the only project in the database, you can simply `DROP` the registry; otherwise, run `sqitch revert --log-only` with the old plan, then you can start deploying from the new.

HTH, and holler if you have any other questions.

**Comment by p-hebert** [comment]
Hey David/@theory!

That's a fantastic answer. Thanks for taking the time to describe the procedure to carry out the squash.
I'll definitely reach out if I need some extra info when I'll carry out the procedure.

Have a wonderful day!
